src/lib/containers/table/table-top/ProgrammaticTableDownload.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useState } from 'react'
+import {
+  QueryBundleRequest,
+  QueryResultBundle,
+} from '../../../utils/synapseTypes'
+import { TransformSqlWithFacetsRequest } from '../../../utils/synapseTypes/Table/TransformSqlWithFacetsRequest'
+import useDeepCompareEffect from 'use-deep-compare-effect'
+import { getTransformSqlWithFacetsRequest } from '../../../utils/SynapseClient'
+import { ProgrammaticInstructionsModal } from '../../ProgrammaticInstructionsModal'
+
+type ProgrammaticOptionsProps = {
+  queryBundleRequest: QueryBundleRequest
+  queryResultBundle: QueryResultBundle
+  onHide: () => void
+}
+
+function ProgrammaticOptions({
+  queryBundleRequest,
+  queryResultBundle,
+  onHide,
+}: ProgrammaticOptionsProps) {
+  const [generatedSql, setGeneratedSql] = useState('')
```

Comment:
Now this component only needs to be responsible for figuring out the code for each programmatic client.